internal/worker/sshsession/worker.go

```diff
@@ -0,0 +1,364 @@
+package sshsession
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/juju/errors"
+	jujussh "github.com/juju/utils/v3/ssh"
+	"github.com/juju/worker/v3"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/sync/errgroup"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/core/watcher"
+)
+
+var (
+	// ControllerSSHUser is the user that will connect to the controller's embedded
+	// SSH server.
+	ControllerSSHUser = "reverse-ssh"
+	// We use authorized_keys2 as it's a default configured alternative file to
+	// authorized_keys. Our existing SSH implementation was hardcoded against authorized_keys
+	// and using this file allows us to not interfere with existing setups. The user may
+	// still add another file to the "AuthorizedKeysFile" directive in sshd_config.
+	//
+	// TODO(ale8k): At somepoint, update juju documentation to explain should they wish
+	// to ssh directly to a machine, this is the way to do so.
+	AuthorizedKeysFile = "authorized_keys2"
+)
+
+// FacadeClient holds the facade methods for the SSH session worker.
+type FacadeClient interface {
+	// WatchSSHConnRequest returns a watcher that will return doc ids for
+	// incoming SSH connection requests. These doc ids are to retrieve connect
+	// params from GetSSHConnRequest(docID string) (SSHConnRequest, error).
+	WatchSSHConnRequest(machineId string) (watcher.StringsWatcher, error)
+
+	// GetSSHConnRequest returns the SSH connection request for the given docID.
+	GetSSHConnRequest(docID string) (DummyParams, error)
```

Comment:
Let's not use DummyParams. We either know the facade interface or we don't. If we don't we should make sure we define it first. We can then choose to implement the worker and mock the facade, or implement the facade first and then the worker second.

Reply:
Discussed with the guys, I've added the real client in this PR (because it's so small and this is the consumer) and removed dummy params.